### ISSUE TITLE
Joblib hijacks the root logger

### ISSUE THREAD

**Issue by mivade** [joblib/joblib#692]
When I clear a cache with `Memory.clear`:

```python
from joblib import Memory

memory = Memory(cachedir=".")
memory.clear()
```

I get output on the root logger which has apparently now been configured with what looks like `logging.basicConfig`:

```
WARNING:root:[Memory(cachedir='./joblib')]: Flushing completely the cache
```

This goes against Python logging best practices because it's manipulating the root logger which propagates to all other loggers. An example of why this is bad practice can be seen below:

```python
import logging
from joblib import Memory

logger = logging.getLogger("mylogger")
logger.setLevel(logging.INFO)
logger.addHandler(logging.StreamHandler())

memory = Memory(cachedir=".")
memory.clear()

logger.info("oops, I'll show up twice")
```

which gives the output

```
WARNING:root:[Memory(cachedir='./joblib')]: Flushing completely the cache
oops, I'll show up twice
INFO:mylogger:oops, I'll show up twice
```

What should be happening instead:

- joblib should *not* use calls to `logging.warning` but instead create its own logger and call `logger.warning` instead
- this logger should add a `NullHandler` so as to not interfere with existing logging
- clearing cache should default to not give a warning in the first place since if I am calling it, I know that it's going to be cleared

**Comment by beetree** [comment]
Still not fixed. Does anyone have a workaround?

**Comment by panaggio** [comment]
Still not fixed. Does anyone have a workaround? [2]

**Comment by joewandy** [comment]
Same problem here. Joblib is messing up all my logging.

**Comment by eromoe** [comment]
Why this still not fix

**Comment by MrSaints** [comment]
This is still an issue.

**Comment by asaph96** [comment]
Any solution to this problem?

**Comment by tae898** [comment]
Still not fixed. Does anyone have a workaround? [3]

**Comment by C0FFEEC0FFEE** [comment]
Here's a workaround:
https://github.com/joblib/joblib/issues/634#issuecomment-367321025

**Comment by amniskin** [comment]
Hey, I just submitted a pull request for this issue. It's #1033. But for some reason a bunch of seemingly unrelated things are now failing (things like pickling io objects). The only file I changed was like 5 lines or so from Logger.py. What gives?

**Comment by scottgigante-immunai** [comment]
This is still an issue. Any chance we can get #1033 merged?

**Comment by tomMoral** [comment]
I just merged #1439 which should provide a solution for this.
Feel free to open a follow-up issue if needed.
